src/ObjectSchema.js

```diff
@@ -323,7 +324,9 @@ const ObjectSchema = ({ schema = initialState, ...options } = {}) => {
           ...schema,
           [target]: [
             ...schema[target],
-            $ref ? { name, $ref } : Object.assign({}, { name }, attributes)
+            $ref
+              ? (Object.keys(restAttributes).length ? { name, $ref, ...restAttributes } : { name, $ref })
+              : Object.assign({}, { name }, attributes)
```

Comment:
I guess we dont need the first empty object if we create the second one on the fly too?

```suggestion
              : Object.assign({ name }, attributes)
```